packages/salesforcedx-vscode-core/src/commands/isvdebugging/bootstrapCmd.ts

```diff
@@ -374,55 +313,35 @@ export class IsvDebugBootstrapExecutor extends SfdxCommandletExecutor<{}> {
     );
     const packageInfos = this.parsePackageInstalledListJson(packagesJson);
 
-    // 6: fetch packages
-    await this.executeCommand(
-      this.buildRetrievePackagesSourceCommand(
-        response.data,
-        packageInfos.map(entry => entry.name)
-      ),
-      { cwd: projectPath },
-      cancellationTokenSource,
-      cancellationToken
-    );
+    // 5a: create directory where packages are to be retrieved
+    shell.mkdir('-p', projectInstalledPackagesPath); // .sfdx/tools/installed-packages
+    const packageNames = packageInfos.map(entry => entry.name);
 
-    // 7a: unzip downloaded packages into temp location
-    try {
-      const packagesTempPath = path.join(projectMetadataTempPath, 'packages');
-      shell.mkdir('-p', packagesTempPath);
-      shell.mkdir('-p', projectInstalledPackagesPath);
-      const zip = new AdmZip(
-        path.join(projectMetadataTempPath, 'unpackaged.zip')
-      );
-      zip.extractAllTo(packagesTempPath, true);
-    } catch (error) {
-      console.error(error);
-      channelService.appendLine(
-        nls.localize('error_extracting_packages', error.toString())
-      );
-      notificationService.showErrorMessage(
-        nls.localize('error_extracting_packages', error.toString())
+    // 5b: retrieve packages
+    // TODO: what if packageNames.length is 0?
+    for (let i = 0; i < (packageNames.length); i++) {
+      await this.executeCommand(
+        this.buildRetrievePackageSourceCommand(
+          response.data,
+          packageNames[i]
+        ),
+        { cwd: projectPath },
+        cancellationTokenSource,
+        cancellationToken
       );
-      return;
     }
 
-    // 7b: convert packages into final location
     for (const packageInfo of packageInfos) {
       channelService.appendLine(
         nls.localize('isv_debug_bootstrap_processing_package', packageInfo.name)
       );
-      await this.executeCommand(
-        this.buildMetadataApiConvertPackageSourceCommand(packageInfo.name),
-        { cwd: projectPath },
-        cancellationTokenSource,
-        cancellationToken
-      );
 
       // generate installed-package.json file
       try {
         fs.writeFileSync(
           path.join(
             projectInstalledPackagesPath,
-            packageInfo.name,
+            packageInfo.name.replace(/[^a-zA-Z0-9]/g, '-'),
```

Comment:
@peternhale It replaces any non alphanumeric character with `'-'` including  __space__. We just want to replace `'.'` with `'-'` .